tests/unit/transformers/test_base.py

```diff
@@ -72,3 +436,191 @@ def test__set_columns_data_preserves_order_dataframe(self):
             'd': [10, 11, 12]
         }, index=[2, 0, 1])
         pd.testing.assert_frame_equal(data, expected)
+
+    def test__build_output_columns(self):
+        """Test the ``_build_output_columns`` method.
+
+        Setup:
+            - set the ``columns`` attribute to a list of a subset of the columns in the data.
+
+        Input:
+            - a dataframe.
+
+        Expected behavior:
+            - ``column_prefix`` should be set to the elements stored in ``columns`` joined by
+            hashtags (e.g. ['a', 'b'] -> 'a#b').
+            - ``output_columns`` should be set to a list of the keys of what's returned from the
+            ``get_output_types`` method.
+        """
+
+    def test__build_output_columns_generated_already_exist(self):
+        """Test the ``_build_output_columns`` method.
+
+        When this method generates column names that already exist in the data, it should
+        keep adding hashtags at the end of the column name, until the column name becomes unique.
+
+        Setup:
+            - set the ``columns`` attribute to a list of a subset of the columns in the data.
+
+        Input:
+            - a dataframe where the generated column name already exist (e.g. ['a', 'b', 'a#b']).
+
+        Expected behavior:
+            - ``column_prefix`` should be set to the elements stored in ``columns`` joined by
+            hashtags, with a hashtag added at the end (e.g. ['a', 'b'] -> 'a#b#').
+            - ``output_columns`` should be set to a list of the keys of what's returned from the
+            ``get_output_types`` method.
+        """
+
+    def test_fit(self):
+        """Test the ``fit`` method.
+
+        Setup:
+            - mock ``_fit``.
```

Comment:
By `mock` you mean `patch`?

Another option is to define a subclass on the fly that implements a `fit` method that stores the input data, so later on you can assert on it:

```python3
def test_fit(self):
    # setup
    class Dummy(BaseTransformer):
        def fit(self, data):
            self._passed_data = data

    # run
    ...
    dummy = Dummy()
    dummy.fit(data)

    # assert
    assert dummy._passed_data is the expected one
```